新增了扩展资源数据库的中英文档 (#1710)
cocos-creator/creator-docs: en/editor/extension/contributions-database.md

> # Extending the Database(DB)

Reviewer: ```suggestion
# Extending the Database (DB)
```

There will be a lot of resources in the project, and the resources will come with a lot of data, which is managed by the resource database.
Extensions can inject resources into the resource database to provide resources like scripts, images, etc.

> Through this article we will learn to inject resources into the resource manager via `contributions` to `asset-db`.

Reviewer: ```suggestion
This article will demonstrate how to inject resources into the resource manager via `contributions` to `asset-db`.
```

## Registration

> Register Database as follows.

Reviewer: ```suggestion
Register the Database as follows:
```

```json5
/// package.json
{
    "name": "test-package",
    "contributions": {
        "asset-db": {
            "mount": {
                "path": "./assets",
                "readonly": true
            }
        }
    }
}
```

```typescript
interface AssetDBConfig {
    mount:{
        //Directory of resources, relative to extension
        path: string；
        //Whether the resource is read-only,read and write by default
        readonly?: boolean;
    }
}
```

## Writing script resources in extensions

> In order to write `ts` script resources in the extension, we need to copy `{project directory}\temp\declarations` to the extension directory.

Reviewer: ```suggestion
In order to write `ts` script resources in the extension, copy `{project directory}\temp\declarations` to the extension directory.
```

> This will give us the `cc` related script definitions.

Reviewer: ```suggestion
This will provide the `cc` related script definitions.
```

## Importing extension injected script resources

> Earlier we created a new extension `test-package`, which injects the resources under the `test-package\assets` path into the resource database.

Reviewer: ```suggestion
Earlier, the documentation demonstrated creating a new extension, `test-package`, which injects the resources under the `test-package\assets` path into the resource database.
```

> Let's assume that the `test-package\assets` path has a script `foo.ts`,

Reviewer: ```suggestion
Assuming that the `test-package\assets` path has a script `foo.ts`:
```

```typescript
/// test-package\assets\foo.ts
export const value = 123;
```

> In the project's script `bar.ts` we should import the script using the following way.

Reviewer: ```suggestion
In the project's script, `bar.ts` should be imported in the following way:
```

# Extending the Database (DB)

There will be a lot of resources in the project, and the resources will come with a lot of data, which is managed by the resource database.
Extensions can inject resources into the resource database to provide resources like scripts, images, etc.
This article will demonstrate how to inject resources into the resource manager via `contributions` to `asset-db`.

## Registration

Register the Database as follows:

```json5
/// package.json
{
    "name": "test-package",
    "contributions": {
        "asset-db": {
            "mount": {
                "path": "./assets",
                "readonly": true
            }
        }
    }
}
```

```typescript
interface AssetDBConfig {
    mount:{
        //Directory of resources, relative to extension
        path: string；
        //Whether the resource is read-only,read and write by default
        readonly?: boolean;
    }
}
```

## Writing script resources in extensions

> We can define the script in the `test-package\assets\` resource folder that we just registered, and we'll start by creating a script `foo.ts` that

Reviewer: ```suggestion
Define the script in the `test-package\assets\` resource folder that was just registered, and start by creating a script `foo.ts` that resembles the following:
```

```typescript
/// foo.ts
import { _decorator, Component, Node } from 'cc';
export const value = 123;
const { ccclass, property } = _decorator;
 
@ccclass('Foo')
export class Foo extends Component {
    start () {
        console.log('foo');
    }
}
```

> > **Note**: In order to use the cc definitions, we need to copy the definition file of `{project directory}\temp\declarations` to the extension directory.

Reviewer: ```suggestion
> **Note**: to use the `cc` definitions, copy the definition file of `{project directory}\temp\declarations` to the extension directory.
```

## Importing extension injected script resources

> Earlier we created a new extension `test-package` that injects the resources under the `test-package\assets` path into the resource database.

Reviewer: ```suggestion
Earlier in the documentation a new extension, `test-package`, was created. It injects the resources under the `test-package\assets` path into the resource database.
```

> In the project's script `bar.ts` we can import the `foo.ts` script in the following way.

Reviewer: ```suggestion
In the project's script `bar.ts` import the `foo.ts` script in the following way.
```
